CMakeLists.txt

```diff
@@ -1,111 +1,104 @@
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.14...3.20)
 
 project(
-    example
-    VERSION 0.1.0
-    LANGUAGES CXX
+  example
+  VERSION 0.1.0
+  LANGUAGES CXX
 )
 
+option(BUILD_SHARED_LIBS "default" YES)
+
 # ---- Warning guard ----
 
 # Protect dependents from this project's warnings if the guard isn't disabled
-set(example_warning_guard SYSTEM)
-if(example_INCLUDE_WITHOUT_SYSTEM)
-  set(example_warning_guard "")
-endif()
+## XXX why? CK
```

Comment:
Imported targets' (ones coming from `find_package`) include paths are implicitly treated as `SYSTEM`. If a user wants to vendor a library, without a warning guard they will have to deal with __your__ warnings in __his__ build tree.

To mirror behavior of imported targets and make users' life easier, this is necessary. If the user still wants warnings, they can be trivially toggled in multiple ways.

Reply:
I don't understand! it is not needed IMHO.

Reply:
With this warning guard dependents can use `add_subdirectory` (ie vendor a dependency) and still get the same behaviour as if they used `find_package`. It's a few lines of code for a very nice convenience function that can be toggled easily. I prefer a happy user to screwing him over with my warnings.